complexity of webhook.ts

### Describe what should be investigated or refactored
```bash
  23:8  warning  Async function 'generateWebhookRules' has a complexity of 11. Maximum allowed is 10  complexity
```

--- a/src/lib/assets/webhooks.ts
+++ b/src/lib/assets/webhooks.ts
@@ -24,40 +24,24 @@ const peprIgnoreNamespaces: string[] = ["kube-system", "pepr-system"];
 const validateRule = (binding: Binding, isMutateWebhook: boolean): V1RuleWithOperations | undefined => {
   const { event, kind, isMutate, isValidate } = binding;
 
-  // If the module doesn't have a callback for the event, skip it
-  if (isMutateWebhook && !isMutate) {
+  // Skip invalid bindings based on webhook type
+  if ((isMutateWebhook && !isMutate) || (!isMutateWebhook && !isValidate)) {
     return undefined;
   }
 
-  if (!isMutateWebhook && !isValidate) {
-    return undefined;
-  }
-
-  const operations: string[] = [];
-
-  // CreateOrUpdate is a Pepr-specific event that is translated to Create and Update
-  if (event === Event.CREATE_OR_UPDATE) {
-    operations.push(Event.CREATE, Event.UPDATE);
-  } else {
-    operations.push(event);
-  }
+  // Translate event to operations
+  const operations = event === Event.CREATE_OR_UPDATE ? [Event.CREATE, Event.UPDATE] : [event];
 
   // Use the plural property if it exists, otherwise use lowercase kind + s
   const resource = kind.plural || `${kind.kind.toLowerCase()}s`;
 
-  const ruleObject = {
+  const ruleObject: V1RuleWithOperations = {
     apiGroups: [kind.group],
     apiVersions: [kind.version || "*"],
     operations,
-    resources: [resource],
+    resources: [resource, ...(resource === "pods" ? ["pods/ephemeralcontainers"] : [])],
   };
 
-  // If the resource is pods, add ephemeralcontainers as well
-  if (resource === "pods") {
-    ruleObject.resources.push("pods/ephemeralcontainers");
-  }
-
-  // Add the rule to the rules array
   return ruleObject;
 };
 

--- a/src/lib/assets/webhooks.ts
+++ b/src/lib/assets/webhooks.ts
@@ -11,6 +11,7 @@ import { concat, equals, uniqWith } from "ramda";
 
 import { Assets } from ".";
 import { Event } from "../enums";
+import { Binding } from "../types";
 
 const peprIgnoreLabel: V1LabelSelectorRequirement = {
   key: "pepr.dev",
@@ -20,9 +21,49 @@ const peprIgnoreLabel: V1LabelSelectorRequirement = {
 
 const peprIgnoreNamespaces: string[] = ["kube-system", "pepr-system"];
 
+const validateRule = (binding: Binding, isMutateWebhook: boolean): V1RuleWithOperations | undefined => {
+  const { event, kind, isMutate, isValidate } = binding;
+
+  // If the module doesn't have a callback for the event, skip it
+  if (isMutateWebhook && !isMutate) {
+    return undefined;
+  }
+
+  if (!isMutateWebhook && !isValidate) {
+    return undefined;
+  }
+
+  const operations: string[] = [];
+
+  // CreateOrUpdate is a Pepr-specific event that is translated to Create and Update
+  if (event === Event.CREATE_OR_UPDATE) {
+    operations.push(Event.CREATE, Event.UPDATE);
+  } else {
+    operations.push(event);
+  }
+
+  // Use the plural property if it exists, otherwise use lowercase kind + s
+  const resource = kind.plural || `${kind.kind.toLowerCase()}s`;
+
+  const ruleObject = {
+    apiGroups: [kind.group],
+    apiVersions: [kind.version || "*"],
+    operations,
+    resources: [resource],
+  };
+
+  // If the resource is pods, add ephemeralcontainers as well
+  if (resource === "pods") {
+    ruleObject.resources.push("pods/ephemeralcontainers");
+  }
+
+  // Add the rule to the rules array
+  return ruleObject;
+};
+
 export async function generateWebhookRules(assets: Assets, isMutateWebhook: boolean): Promise<V1RuleWithOperations[]> {
-  const { config, capabilities } = assets;
   const rules: V1RuleWithOperations[] = [];
+  const { config, capabilities } = assets;
 
   // Iterate through the capabilities and generate the rules
   for (const capability of capabilities) {
@@ -30,43 +71,10 @@ export async function generateWebhookRules(assets: Assets, isMutateWebhook: bool
 
     // Read the bindings and generate the rules
     for (const binding of capability.bindings) {
-      const { event, kind, isMutate, isValidate } = binding;
-
-      // If the module doesn't have a callback for the event, skip it
-      if (isMutateWebhook && !isMutate) {
-        continue;
+      const validatedRule = validateRule(binding, isMutateWebhook);
+      if (validatedRule) {
+        rules.push(validatedRule);
       }
-
-      if (!isMutateWebhook && !isValidate) {
-        continue;
-      }
-
-      const operations: string[] = [];
-
-      // CreateOrUpdate is a Pepr-specific event that is translated to Create and Update
-      if (event === Event.CREATE_OR_UPDATE) {
-        operations.push(Event.CREATE, Event.UPDATE);
-      } else {
-        operations.push(event);
-      }
-
-      // Use the plural property if it exists, otherwise use lowercase kind + s
-      const resource = kind.plural || `${kind.kind.toLowerCase()}s`;
-
-      const ruleObject = {
-        apiGroups: [kind.group],
-        apiVersions: [kind.version || "*"],
-        operations,
-        resources: [resource],
-      };
-
-      // If the resource is pods, add ephemeralcontainers as well
-      if (resource === "pods") {
-        ruleObject.resources.push("pods/ephemeralcontainers");
-      }
-
-      // Add the rule to the rules array
-      rules.push(ruleObject);
     }
   }
 

--- a/src/lib/assets/webhooks.ts
+++ b/src/lib/assets/webhooks.ts
@@ -62,23 +62,16 @@ const validateRule = (binding: Binding, isMutateWebhook: boolean): V1RuleWithOpe
 };
 
 export async function generateWebhookRules(assets: Assets, isMutateWebhook: boolean): Promise<V1RuleWithOperations[]> {
-  const rules: V1RuleWithOperations[] = [];
   const { config, capabilities } = assets;
 
-  // Iterate through the capabilities and generate the rules
-  for (const capability of capabilities) {
+  const rules = capabilities.flatMap(capability => {
     console.info(`Module ${config.uuid} has capability: ${capability.name}`);
 
-    // Read the bindings and generate the rules
-    for (const binding of capability.bindings) {
-      const validatedRule = validateRule(binding, isMutateWebhook);
-      if (validatedRule) {
-        rules.push(validatedRule);
-      }
-    }
-  }
+    return capability.bindings
+      .map(binding => validateRule(binding, isMutateWebhook))
+      .filter((rule): rule is V1RuleWithOperations => !!rule);
+  });
 
-  // Return the rules with duplicates removed
   return uniqWith(equals, rules);
 }
 

